libs/user_extensions/src/TTAlpsHistogramFiller.cpp

```diff
@@ -671,8 +670,33 @@ void TTAlpsHistogramFiller::FillLLPnanoAODLooseMuonsVertexHistograms(const share
 }
 
 void TTAlpsHistogramFiller::FillLLPnanoAODLooseMuonsNminus1VertexHistograms(const shared_ptr<Event> event){
+  float weight = GetEventWeight(event);
+
   for(auto collectionName : muonVertexNminus1Collections) {
-    FillMuonVertexHistograms(event, collectionName);
+    // Only for the BestVertex collections
+    if(collectionName.find("Best") == string::npos) continue;
+    for(auto cut : muonVertexCollections[collectionName]) {
+
+      if(cut.find("Cut") == string::npos) continue;
```

Comment:
I'm confused by what "cut" is here, given it comes from the muon vertex collection. I guess there may be some naming issue here. Also, what does the "if" statement do? Maybe this line deserves a short comment